sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java

```diff
@@ -170,11 +173,13 @@ public WriteResult expand(PCollection<KV<DestinationT, TableRow>> input) {
                     new DoFn<Void, String>() {
                       @ProcessElement
                       public void getTempFilePrefix(ProcessContext c) {
-                        c.output(
-                            resolveTempLocation(
-                                c.getPipelineOptions().getTempLocation(),
-                                "BigQueryWriteTemp",
-                                stepUuid));
+                        String tempLocation = resolveTempLocation(
+                            c.getPipelineOptions().getTempLocation(),
+                            "BigQueryWriteTemp",
+                            BigQueryHelpers.randomUUIDString());
```

Comment:
Yeah - no reason for it to be except for visibility, and we are logging the tempLocation so that shouldn't be an issue.